packages/inventory-compliance/src/Components/RulesTable/index.js

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import propTypes from 'prop-types';
+import { Pagination, PaginationVariant } from '@patternfly/react-core';
+import { AnsibeTowerIcon } from '@patternfly/react-icons';
+import { Table, TableBody, TableHeader, fitContent } from '@patternfly/react-table';
+import TableToolbar from '@redhat-cloud-services/frontend-components/TableToolbar';
+import PrimaryToolbar from '@redhat-cloud-services/frontend-components/PrimaryToolbar';
+import useTableTools from '@redhat-cloud-services/frontend-components-utilities/useTableTools';
+import ComplianceRemediationButton from '../../ComplianceRemediationButton';
+import { Rule, Policy, Severity, Passed, Ansible } from './Cells';
+import RuleDetailsRow from './RuleDetailsRow';
+import EmptyRows from './EmptyRows';
+import buildFilterConfig from './Filters';
+
+/* eslint-disable react/display-name */
+export const columns = [
+    {
+        title: 'Name',
+        prop: 'title',
+        sortByProperty: 'title',
+        renderFunc: (rule) => (
+            <Rule { ...{ rule } } />
+        )
+    }, {
+        title: 'Policy',
+        sortByFunction: (rule) => (rule?.profile?.name),
+        renderFunc: (rule) => (
+            <Policy { ...{ rule } } />
+        )
+    }, {
+        title: 'Severity',
+        transforms: [ fitContent ],
+        sortByProperty: 'severity',
+        sortByArray: [ 'high', 'medium', 'low', 'unknown' ],
+        renderFunc: (rule) => (
+            <Severity { ...{ rule } } />
+        )
+    }, {
+        title: 'Passed',
+        transforms: [ fitContent ],
+        sortByProperty: 'compliant',
+        renderFunc: (rule) => (
+            <Passed { ...{ rule } } />
+        )
+    }, {
+        title: <React.Fragment><AnsibeTowerIcon /> Ansible</React.Fragment>,
+        original: 'Ansible',
+        prop: 'remediationAvailable',
+        props: { tooltip: 'Ansible' },
+        transforms: [ fitContent ],
+        sortByProperty: 'remediationAvailable',
+        renderFunc: (rule) => (
+            <Ansible { ...{ rule } } />
+        )
+    }
+];
+/* eslint-enable */
+
+export const selectColumns = (columnTitles) => (
+    columns.filter((column) => columnTitles.includes(column.original || column.title))
```

Comment:
Observation: There is a column matching by `original` and `title` used through the code. Could it use one kind, having `original` as primary? Preferably the `original` could be named `name` to provide a programmatic naming to a column.

Currently the column matching is mostly coupled with a visual title. If the title is changed (as we've seen with "Rule" -> "Name") the whole usage has to change with it. Having a single and stable naming would prevent visual changes affecting the column selection/matching.

Reply:
Yeah, @akofink pointed these out as well, somewhere similar, I can't remember where, but essentially we'll move from using a string to have cell object to import and pass in in the place where it is used, this will also allow to extend these "cell objects" as needed with custom props and `selectColumn` will be obsolete.